sdk/src/types/block/context_input/mod.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod reward;
+
+use derive_more::From;
+
+use self::reward::RewardContextInput;
+use crate::types::block::Error;
+
+/// A Context Input provides additional contextual information for the execution of a transaction, such as for different
+/// functionality related to accounts, commitments, or Mana rewards. A Context Input does not need to be unlocked.
+#[derive(Clone, Eq, PartialEq, Hash, Ord, PartialOrd, From, packable::Packable)]
+#[packable(unpack_error = Error)]
+#[packable(tag_type = u8, with_error = Error::InvalidInputKind)]
```

Comment:
I know they are similar, but maybe it should have its own error for better debugging

Reply:
Updated